test/unit/LimesTests.ts

```diff
@@ -342,4 +341,136 @@ suite('Limes', (): void => {
       });
     });
   });
+
+  suite('verifyTokenInWebsocketUpgradeRequest', (): void => {
+    test('returns an anonymous token for non-authenticated requests.', async (): Promise<void> => {
+      const upgradeRequest: Request = {
+        headers: {}
+      } as Request;
+
+      const result = await limes.verifyTokenInWebsocketUpgradeRequest({
+        issuerForAnonymousTokens: 'https://untrusted.thenativeweb.io',
+        upgradeRequest
+      });
+
+      assert.that(result.token).is.startingWith('ey');
+      assert.that(result.user.id).is.equalTo('anonymous');
+      assert.that(result.user.claims.sub).is.equalTo('anonymous');
+      assert.that(result.user.claims.iss).is.equalTo('https://untrusted.thenativeweb.io');
+      assert.that(result.user.claims['https://untrusted.thenativeweb.io/is-anonymous']).is.true();
+    });
+
+    test('returns an anonymous with the provided id for non-authenticated requests.', async (): Promise<void> => {
+      const anonymousId = uuid();
+      const upgradeRequest: Request = {
+        headers: {
+          'x-anonymous-id': anonymousId
+        }
+      } as unknown as Request;
+
+      const result = await limes.verifyTokenInWebsocketUpgradeRequest({
+        issuerForAnonymousTokens: 'https://untrusted.thenativeweb.io',
+        upgradeRequest
+      });
+
+      assert.that(result.token).is.startingWith('ey');
+      assert.that(result.user.id).is.equalTo(`anonymous-${anonymousId}`);
+      assert.that(result.user.claims.sub).is.equalTo(`anonymous-${anonymousId}`);
+      assert.that(result.user.claims.iss).is.equalTo('https://untrusted.thenativeweb.io');
+      assert.that(result.user.claims['https://untrusted.thenativeweb.io/is-anonymous']).is.true();
+    });
+
+    test('throws an error for invalid tokens.', async (): Promise<void> => {
+      const upgradeRequest: Request = {
+        headers: {
+          authorization: 'Bearer invalidtoken'
+        }
+      } as unknown as Request;
+
+      await assert.that(async (): Promise<any> => await limes.verifyTokenInWebsocketUpgradeRequest({
+        issuerForAnonymousTokens: 'https://untrusted.thenativeweb.io',
+        upgradeRequest
+      })).is.throwingAsync();
```

Comment:
Since we don't expect anything back here (in contrast: since we expect it to fail, it should never return anything), I'd suggest to use `Promise<void>` here and add curly braces.

```suggestion
      await assert.that(async (): Promise<void> => {
        await limes.verifyTokenInWebsocketUpgradeRequest({
          issuerForAnonymousTokens: 'https://untrusted.thenativeweb.io',
          upgradeRequest
        });
      }).is.throwingAsync();
```